bin/i18n/i18n_script_utils.rb

```diff
@@ -55,10 +55,17 @@ def self.to_crowdin_yaml(data)
 
     # Make sure we treat the strings 'y' and 'n' as strings, and not bools
     yaml_bool = /^(?:y|Y|n|N)$/
+    # Make sure we use the right format for strings with '#' in them, otherwise
+    # YAML parsers might treat part of the string as a YAML comment.
+    octothorpe = /#/
     ast.grep(Psych::Nodes::Scalar).each do |node|
       if yaml_bool.match node.value
         node.plain = false
         node.quoted = true
+      elsif octothorpe.match node.value
```

Comment:
Do we need a separate block for '#' strings if we're doing the same thing to them as the booleans?

Reply:
I agree that they could probably be merged. For this PR, I wanted to focus on minimizing changes since we have been struggling to get an i18n sync out.